volttron/platform/store.py

```diff
@@ -284,9 +233,40 @@ def set_config(self, config_name, contents, trigger_callback=False):
 
     @RPC.export
     def get_configs(self):
-        """Called by an Agent at startup to get the initial configuration state."""
+        """Called by an Agent at startup to trigger initial configuration state push."""
         identity = bytes(self.vip.rpc.context.vip_message.peer)
-        return self.store.get(identity, {}).get("configs", {})
+
+        #We need to create store and lock if it doesn't exist in case someone tries to add
+        # a configuration while we are sending the initial state.
+        agent_store = self.store.get(identity)
+
+
+        if agent_store is None:
+            # Initialize a new store.
+            store_path = os.path.join(self.store_path, identity + store_ext)
+            store = PersistentDict(filename=store_path, flag='c', format='json')
```

Comment:
Yeah, right now it will probably blowup. Should only affect the bad agent and not cause any lasting harm.